### PR TITLE
Fix transition duplicate TU and fix End-Use Subcategory value

### LINKED ISSUE #10746: Transition created duplicate ZoneHVAC:TerminalUnit:VariableRefrigerantFlow object
Issue overview
--------------
The v24.1 to 24.2 transition code will create one more copy of `ZoneHVAC:TerminalUnit:VariableRefrigerantFlow` when the `Supply Air Fan Object Type` is already `Fan:SystemModel`

### Details
Some additional details for this issue (if relevant):
 - Platform (Mac 13.5.2)
 - Version of EnergyPlus (24.2)
 - Unmethours link or helpdesk ticket number
 - defect file
[US+SF+CZ5B+hp+slab+IECC_2021_VRFPhysics_v2_hrdsize_V2310.idf.zip](https://github.com/user-attachments/files/17050248/US%2BSF%2BCZ5B%2Bhp%2Bslab%2BIECC_2021_VRFPhysics_v2_hrdsize_V2310.idf.zip)

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added (list location of defect file here)
 - [ ] Ticket added to EnergyPlus Defect Complexity (Github Project)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by yujiex**
Pull request overview
---------------------
 - Fixes #10746
 - Fixes the transition problem where a duplicate terminal unit (`ZoneHVAC:TerminalUnit:VariableRefrigerantFlow`) object is created in the transition
 - Also fixes a problem where the "End-Use Subcategory" field is not correctly initialized when the `ZoneHVAC:TerminalUnit:VariableRefrigerantFlow` has `Fan:VariableVolume` in `Supply Air Fan Object Type`.

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/Transition/CreateNewIDFUsingRulesV24_2_0.f90 [hunk 1]
```diff
@@ -394,8 +394,10 @@ SUBROUTINE CreateNewIDFUsingRules(EndOfFile,DiffOnly,InLfn,AskForInput,InputFile
               OldFanVO(NumOldFanVO)%coeff5 = TRIM(IDFRecords(Num)%Numbers(12))             !- Coefficient5 x**4
               OldFanVO(NumOldFanVO)%inletAirNodeName = TRIM(IDFRecords(Num)%Alphas(4))
               OldFanVO(NumOldFanVO)%outletAirNodeName = TRIM(IDFRecords(Num)%Alphas(5))
-              IF (lEN(IDFRecords(Num)%Alphas) .eq. 6) THEN
+              IF (SIZE(IDFRecords(Num)%Alphas) .eq. 6) THEN
```
Comment (Myoldmopar) [inline src/Transition/CreateNewIDFUsingRulesV24_2_0.f90:397]: Ahh good fix.  Yeah, LEN on an CHARACTER ARRAY returns the length of each string inside the array, not the size of the array itself.  Subtle.

**Review by Myoldmopar — APPROVED** [review]
:+1:

**Comment by Myoldmopar** [comment]
Good to go, thanks for the transition fix.

**Comment by yujiex** [comment]
Thanks @Myoldmopar

**Pull request merged**